packages/data-stores/src/reader/mutations/use-user-settings-mutation.ts

```diff
@@ -34,31 +35,29 @@ const useUserSettingsMutation = () => {
 		},
 		{
 			onMutate: async ( data ) => {
-				await queryClient.cancelQueries( [ 'read', 'email-settings', isLoggedIn ] );
-				const previousSettings = queryClient.getQueryData< SubscriptionManagerUserSettings >( [
-					'read',
-					'email-settings',
-					isLoggedIn,
-				] );
+				await queryClient.cancelQueries( emailSettingsCacheKey );
+				const previousSettings =
+					queryClient.getQueryData< SubscriptionManagerUserSettings >( emailSettingsCacheKey );
 
-				queryClient.setQueryData< SubscriptionManagerUserSettings >(
-					[ 'read', 'email-settings', isLoggedIn ],
-					{ ...previousSettings, ...data }
-				);
+				queryClient.setQueryData< SubscriptionManagerUserSettings >( [ 'read', 'email-settings' ], {
```

Comment:
I think you meant to use the `emailSettingsCacheKey` instead of `[ 'read', 'email-settings' ]` as the key in this cache update? 🤔

Reply:
Yes, nice catch!